SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -186,19 +167,97 @@ QueryConstraint InterestFactory::CreateSystemDefinedConstraints()
 	return SystemDefinedConstraints;
 }
 
-QueryConstraint InterestFactory::CreateUserDefinedConstraints()
+QueryConstraint InterestFactory::CreateCheckoutRadiusConstraints()
 {
-	return QueryConstraint{};
-}
+	// Checkout Radius constraints are defined by the NetCullDistanceSquared property on actors.
+	//   - Checkout radius is a RelativeCylinder constraint on the player controller.
+	//   - NetCullDistanceSquared on AActor is used to define the default checkout radius with no other constraints.
+	//   - NetCullDistanceSquared on other actor types is used to define additional constraints if needed.
+	//   - If a subtype defines a radius smaller than a parent type, then its requirements are already captured.
+	//   - If a subtype defines a radius larger than all parent types, then it needs an additional constraint.
+	//   - Other than the default from AActor, all radius constraints also include Component constraints to
+	//     capture specific types, including all derived types of that actor.
+	//
+	// For best performance, the AActor NetCullDistanceSquared should be set to a value that captures the
+	// majority of actors important to a client, and actors outside that distance should be included by
+	// increasing the NetCullDistanceSquared property on that actor (or a parent actor type to capture a
+	// hierarchy). This should result in a minimal set of constraints for the default query.
+
+	const AActor* DefaultActor = Cast<AActor>(AActor::StaticClass()->GetDefaultObject());
+	const float DefaultDistanceSquared = DefaultActor->NetCullDistanceSquared;
+
+	// Gather NetCullDistance settings, and add any larger than the default radius to a list for processing.
+	TMap<UClass*, float> ActorCullDistances;
+	for (TObjectIterator<UClass> It; It; ++It)
```

Comment:
Would TActorIterator work here and save the cast + check?

Reply:
no, since that gives actor instances. I want uclass instances.